bin/exit_code.ml

```diff
@@ -0,0 +1,18 @@
+type t =
+  | Success
+  | Error
+  | Signal
+
+let all = [ Success; Error; Signal ]
+
+let code = function
+  | Success -> 0
+  | Error -> 1
+  | Signal -> 130
+
+let doc = function
+  | Success -> "on success"
+  | Error -> "if an error happened"
+  | Signal -> "if it was interrupted by a signal"
```

Comment:
Actually I would prefer
```suggestion
  | Success -> "On success."
  | Error -> "If an error happened."
  | Signal -> "If it was interrupted by a signal."
```
And we should document what happens with code_error.
At the moment it looks like:
```
EXIT STATUS
       build exits with the following status:

       0   on success

       1   if an error happened

       130 if it was interrupted by a signal
```
Which I think looks kind of poor.

Reply:
the defaults for cmdliner use lowercase, which I read as a sentence:

- 0 on success
- 1 if an error happened
- 130 it was interrupted by a signal

Reply:
I do agree about the dots though (once again my north star with that is to be consistent with the cmdliner docs)